src/Catalog/Monitoring/Validation/Test/ValidationTestRunnerFactory.cs

```diff
@@ -0,0 +1,111 @@
+﻿using NuGet.Protocol.Core.Types;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using NuGet.Protocol;
+using System.Reflection;
+
+namespace NuGet.Services.Metadata.Catalog.Monitoring.Validation.Test
+{
+    /// <summary>
+    /// Creates <see cref="IValidationTestRunner"/>s using <see cref="SourceRepository"/>s.
+    /// </summary>
+    public class ValidationTestRunnerFactory
+    {
+        /// <summary>
+        /// Exception wrapper for unexpected exceptions thrown during initialization of <see cref="IValidationTestRunner"/>s.
+        /// </summary>
+        public class ValidationTestRunnerFactoryException : Exception
```

Comment:
This type exists so that the `ValidationTestRunner` knows that the test failed due to failed validation and not a transient exception. I used to have a comment that explained this, but basically, if the exception thrown is not a `ValidationTestRunnerFactoryException`, then the `ValidationTestRunner` tries the test again.